packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -477,10 +474,23 @@ class XcodeProjectInfo {
     }
     return false;
   }
+
   /// Returns unique scheme matching [buildInfo], or null, if there is no unique
   /// best match.
-  String? schemeFor(BuildInfo? buildInfo) {
-    final String expectedScheme = expectedSchemeFor(buildInfo);
+  String? schemeFor(String hostAppProjectName, {String? flavor}) {
```

Comment:
Suggest keeping `hostAppProjectName` as a named parameter as you had before, but mark it required:
```suggestion
  String? schemeFor({required String hostAppProjectName, String? flavor}) {
```